test/to_vec.jl

```diff
@@ -200,6 +200,13 @@ end
         test_to_vec(Vector; check_inferred=false) # isa UnionAll
     end
 
+    @testset "CartesianIndex" begin
+        test_to_vec(CartesianIndex(1); check_inferred=false)
```

Comment:
Is it obvious why this doesn't infer?

Reply:
I can't reproduce the inference failure any more 🤷